server/src/DBTablesMonitoring.cc

```diff
@@ -2101,8 +2099,16 @@ void DBTablesMonitoring::updateHosts(const HostInfoList &hostInfoList,
 	HostInfoListConstIterator newHostsItr = hostInfoList.begin();
 	for (; newHostsItr != hostInfoList.end(); ++newHostsItr) {
 		const HostInfo &newHostInfo = *newHostsItr;
+		HostIdHostInfoMapConstIterator currHostItr = currValidHosts.find(newHostInfo.id);
+		if (currHostItr != currValidHosts.end()) {
+			const HostInfo &existHost = *currHostItr->second;
+			if (existHost.hostName != newHostInfo.hostName){
+				updatedHostInfoList.push_back(newHostInfo);
+				continue;
+			}
+		}
 		if (currValidHosts.erase(newHostInfo.id) >= 1) {
```

Comment:
This code does the search twice (find() and erase()) when an element in currValidHosts is removed.
erase() accepts an iterator that is directly pointing an element.

I.e.
currHostItr should be passed to erase() when it should be removed.